test/Serilog.Tests/Support/TemporarySelfLog.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System;
+using System.Collections.Generic;
+using Serilog.Debugging;
+
+namespace Serilog.Tests.Support
+{
+    public class TemporarySelfLog : IDisposable
```

Comment:
We've needed something like this for a while 👍 

(A future version might modify this slightly to only siphon off messages for one logical call context/thread at a time, as this will have interleaving problems when we introduce more tests using it that might run in parallel.)

Reply:
Not quite sure what that would look like... `ThreadStatic` comes to mind, but not sure that would help in that case.